test/test_multiprocessing.py

```diff
@@ -418,9 +433,56 @@ def test_variable_sharing(self):
                            requires_grad=requires_grad)
             self._test_autograd_sharing(var)
 
+    def test_leaf_variable_sharing(self):
+        devices = ['cpu']
+        if torch.cuda.is_available() and not NO_MULTIPROCESSING_SPAWN and TEST_CUDA_IPC:
+            devices.append('cuda')
+        for device in devices:
+            for requires_grad in [True, False]:
+                var = Variable(torch.arange(1., 26, device=device).view(5, 5), requires_grad=requires_grad)
+                self.assertTrue(var.is_leaf)
+                ctx = mp.get_context('spawn') if device == 'cuda' else mp
+                ready = ctx.Event()
+                queue = ctx.Queue()
+                p = ctx.Process(target=requires_grad_variable_sharing, args=(queue, ready))
+                p.daemon = True
+                p.start()
+                queue.put(var)
+                ready.wait()
+                worker_requires_grad = queue.get()
+                self.assertTrue(worker_requires_grad == requires_grad)
+
+    def test_non_leaf_variable_sharing(self):
+        devices = ['cpu'] if not torch.cuda.is_available() else ['cpu', 'cuda']
+        for device in devices:
+            var0 = Variable(torch.arange(1., 26, device=device).view(5, 5), requires_grad=True)
+            var = var0 * 2
+            # We can't do the pickling indirectly, e.g., with a queue.put,
```

Comment:
You can use `SimpleQueue` which pickles immediately. (Queue uses a background thread)